### DISCUSSION TITLE
Conflict Solver Order of Operations

### DISCUSSION THREAD

**Discussion by DiettingerX** [bcssov/IronyModManager#243] [Ideas]
Just an observation:
I find that when I am resolving conflicts between Stellaris mods that if I resolve the Invalid first and run the conflict solver again it provides a much better set of outcomes.

Perhaps moving the invalid list to the top of the categories to be resolved is a good idea.

**Comment by bcssov** [comment]
This largely seems to be inspired by a personal preference one which might seem small but can be annoying to many who have gotten used to invalids being at the bottom. Do you have any use cases where you see the benefit of this change?

**Reply by DiettingerX** [reply]
What I have found is that a number of the mods I am integrating have errors with closing arguments with "}".
The result is that IMM interprets them either incorrectly or simply shunts them aside.
By fixing those errors first in the actual mods and then running the conflict manager a second time they are correctly interpreted by IMM.

This is definitely a low priority unless it is dead simple to fix... but it would be helpful for new modders.

**Comment by bcssov** [comment]
Irony doesn't shunt them aside it cannot parse them properly because they are faulty, and some of these game cannot parse as well. For newbies this will be even more confusing if they are presented with please fix me stuff which requires technical knowledge. Sure so does merging but it's whole of a different game to merge stuff and to code stuff. 

So I'd say no to this.
